### PR TITLE
[Windows] Align Button VisualStates behavior with other platforms

### LINKED ISSUE #7570: "Pressed" Visual State not changing back to "Normal" until you click away on Windows
### Description

I started a new MAUI app to see how to change a button's visual state when it has been pressed. I found that the code I used for this works on macOS but not on Windows. 

Firstly, I added colours to the Colors.xaml file
```
<Color x:Key="PrimaryBackgroundColour">#337ab7</Color>
<Color x:Key="PrimaryBorderColour">#2e6da4</Color>
<Color x:Key="PrimaryBackgroundActiveColour">#204d74</Color>
<Color x:Key="PrimaryBorderActiveColour">#122b40</Color>
```

I then added a new style to the Styles.xaml file which adds the Visual States
```
<Style x:Key="PrimaryButton"  
        TargetType="Button">
    <Setter Property="VisualStateManager.VisualStateGroups">
        <VisualStateGroupList>
            <VisualStateGroup x:Name="CommonStates">
                <VisualState x:Name="Normal">
                    <VisualState.Setters>
                        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource PrimaryBackgroundColour}, Dark={StaticResource White}}" />
                        <Setter Property="BorderColor" Value="{AppThemeBinding Light={StaticResource PrimaryBorderColour}, Dark={StaticResource White}}" />
                    </VisualState.Setters>
                </VisualState>
                <VisualState x:Name="Pressed">
                    <VisualState.Setters>
                        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource PrimaryBackgroundActiveColour}, Dark={StaticResource White}}" />
                        <Setter Property="BorderColor" Value="{AppThemeBinding Light={StaticResource PrimaryBorderActiveColour}, Dark={StaticResource White}}" />
                    </VisualState.Setters>
                </VisualState>
            </VisualStateGroup>
        </VisualStateGroupList>
    </Setter>
</Style>
```

I then changed by Button to have this new style:
```
<Button 
    Text="Login"
    WidthRequest="300"
    HorizontalOptions="Center"
    Style="{StaticResource PrimaryButton}" />
```

When I ran the app on Windows I found that the button's Visual State when pressed did change correctly, but it didn't revert back to the Normal state until I clicked away (for example, clicking in to one of the Entry controls):
Example on Windows

When I run the same code on macOS it works as expected:
Example of macOS

### Steps to Reproduce

I create a new Maui app. 

I added colours to the Colors.xaml file
```
<Color x:Key="PrimaryBackgroundColour">#337ab7</Color>
<Color x:Key="PrimaryBorderColour">#2e6da4</Color>
<Color x:Key="PrimaryBackgroundActiveColour">#204d74</Color>
<Color x:Key="PrimaryBorderActiveColour">#122b40</Color>
```

I then added a new style to the Styles.xaml file which adds the Visual States
```
<Style x:Key="PrimaryButton"  
        TargetType="Button">
    <Setter Property="VisualStateManager.VisualStateGroups">
        <VisualStateGroupList>
            <VisualStateGroup x:Name="CommonStates">
                <VisualState x:Name="Normal">
                    <VisualState.Setters>
                        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource PrimaryBackgroundColour}, Dark={StaticResource White}}" />
                        <Setter Property="BorderColor" Value="{AppThemeBinding Light={StaticResource PrimaryBorderColour}, Dark={StaticResource White}}" />
                    </VisualState.Setters>
                </VisualState>
                <VisualState x:Name="Pressed">
                    <VisualState.Setters>
                        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource PrimaryBackgroundActiveColour}, Dark={StaticResource White}}" />
                        <Setter Property="BorderColor" Value="{AppThemeBinding Light={StaticResource PrimaryBorderActiveColour}, Dark={StaticResource White}}" />
                    </VisualState.Setters>
                </VisualState>
            </VisualStateGroup>
        </VisualStateGroupList>
    </Setter>
</Style>
```

I then changed by Button to have this new style:
```
<Button 
    Text="Login"
    WidthRequest="300"
    HorizontalOptions="Center"
    Style="{StaticResource PrimaryButton}" />
```

### Version with bug

6.0 Release Candidate 3

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10.0.19041.0

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Align Windows Button VisualStates behavior with other platforms.

test-button-pressed

We keep the possibility of setting the focus with Keyboard, etc.
test-button-focus

### Issues Fixed

Fixes #7570

### CODE REVIEW

File: src/Core/src/Handlers/Button/ButtonHandler.Windows.cs [hunk 1]
```diff
@@ -108,6 +111,24 @@ void OnPointerPressed(object sender, PointerRoutedEventArgs e)
...
+			// while on the rest of supported platforms, it goes to Normal.
+			// Here we force the same behavior as in the rest of the platforms but we keep the possibility of setting the focus with Keyboard, etc.
+			UpdateVisualState();
+		}
+
+		void UpdateVisualState()
+		{
+			if (PlatformView is not null)
+			{
+				var dispatcher = MauiContext?.Services.GetRequiredService<IDispatcher>();
```
Comment (mandel-macaque) [inline src/Core/src/Handlers/Button/ButtonHandler.Windows.cs:125]: If MauiContext is null, the we have Services as null and then GetRequiredService will throw a NRE, correct?

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Button/ButtonHandler.Windows.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/Button/ButtonHandler.Windows.cs]: Right, added protection to this posible case.

**Review by rachelkang — COMMENTED** [review]
I think this is a design choice we should have more discussion about. cc @PureWeen @StephaneDelcroix 

This addresses an issue that has come up a few times before. We observed that the current behavior in .NET MAUI is in-line with the Windows platform behavior. A Focused or PointerOver state needed to be added (as you also commented in the original issue) in order to offset the Pressed state. We knew it wasn't the most intuitive thing, so we updated our templates styles to guide developers; maybe we should update the docs for further guidance as well?

I'm concerned about the implications that changing this might have on Focus (not sure, as I haven't tested yet) and on the interaction with Focus and PointerOver states in general. And even if we do decide to update the logic so that the state automatically reverts to the Normal state OnPointerReleased, I do think we would at the very least need some additional checks surrounding focus

**Review by mandel-macaque — APPROVED** [review]

**Comment by PureWeen** [comment]
> I think this is a design choice we should have more discussion about. cc @PureWeen @StephaneDelcroix
> 
> This addresses an issue that has come up a few times before. We observed that the current behavior in .NET MAUI is in-line with the Windows platform behavior. A Focused or PointerOver state needed to be added (as you also commented in the original issue) in order to offset the Pressed state. We knew it wasn't the most intuitive thing, so we updated our templates styles to guide developers; maybe we should update the docs for further guidance as well?
> 
> I'm concerned about the implications that changing this might have on Focus (not sure, as I haven't tested yet) and on the interaction with Focus and PointerOver states in general. And even if we do decide to update the logic so that the state automatically reverts to the Normal state OnPointerReleased, I do think we would at the very least need some additional checks surrounding focus

I agree with Rachel here. I think this is a continuation of my comment here https://github.com/dotnet/maui/pull/10782#issuecomment-1537633102

I feel like we should be fixing these things at the xplat level and not trying to force non standard behavior. If controls all had default styles/default vsms then this wouldn't really be an issue.

**Pull request closed without merging**